core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/config/BaseSailConfig.java

```diff
@@ -80,4 +95,18 @@ public void parse(Model graph, Resource implNode) throws SailConfigException {
 			throw new SailConfigException(e.getMessage(), e);
 		}
 	}
+
+	/**
+	 * @return the defaultQueryEvaluationMode
+	 */
+	public Optional<QueryEvaluationMode> getDefaultQueryEvaluationMode() {
+		return Optional.ofNullable(defaultQueryEvaluationMode);
```

Comment:
Do you mean other methods in `BaseSailConfig`? I'm not entirely sure to be honest. I went with optional because I like it as a design principle - but if the same class has other existing methods that don't use optional yet may return `null` then perhaps you're right that we should change this, for consistency.

Reply:
There are two other getters in this class that don't use optional yet can return `null`, but I'm going to leave as-is on the principle that we want incremental improvement - and using Optional to be explicit about nullability feels like an improvement to me.